pkg/lib/timeformat.ts

```diff
@@ -8,47 +8,60 @@ import { parse, formatDistanceToNow } from 'date-fns';
 import * as locales from 'date-fns/locale';
 
 // this needs to be dynamic, as some pages don't initialize cockpit.language right away
-export const dateFormatLang = () => cockpit.language.replace('_', '-');
+export const dateFormatLang = (): string => cockpit.language.replace('_', '-');
```

Comment:
Was it necessary to annotate the return type?

In general, return types can be omitted, and I usually do that.  On the other hand, I've seen noises from "smart people" (including the TypeScript team) which suggest that type inference in other parts of the program will be faster if you explicitly annotate...

This isn't a request for change — just random musings.

Reply:
I don't think it was *necessary* -- but if we do this, IMHO we should explicitly type all external API, to make it more self-documenting and avoid ambiguity. I like the python mypy approach to require types in declarations, even if inference could figure them out.

Reply:
I'm going to start writing explicit return types from now on.